File: Лабы/ЛР_3/Matlab/division_polinom.m
function [res1,res2,res] = division_polinom(p1,p2)
    res = p1;
	res1 = [];
	res2 = p1;
    d = -1*p2([2:numel(p2)]); 
	if(numel(p1)>=numel(p2))
		for i = [1:numel(p1)-numel(d)]
			res(i) = res(i)/p2(1);
			for j = [1:numel(d)]
				res(i+j) = res(i+j)+res(i)*d(j);
            end
        end
        res1 = res([1:numel(p1)-numel(d)]);
        res2 = res([numel(p1)-numel(d)+1:numel(p1)]);
    end
end

File: Лабы/ЛР_3/Python/division_polinom.py
def division_polinom(p1,p2):
	res = [i for i in p1]
	res1 = []
	res2 = [i for i in p1]
	d = [-p2[i] for i in range(1,len(p2))]
	if(len(p1)>=len(p2)):
		for i in range(0,len(p1)-len(d)):
			res[i] /= p2[0]
			for j in range(len(d)):
				res[i+1+j] += res[i]*d[j]
		res1 = [res[i] for i in range(len(p1)-len(d))]
		res2 = [res[i] for i in range(len(p1)-len(d),len(p1))]
	print(res1,res2,":",res)
	return res

division_polinom([-1,-2,-2,-5,0,-3,-4,2,-5,0],[1,4,1,-14,-20,-8])
division_polinom([1,4,1,-14,-20,-8],[1,3])
division_polinom([1,0,0,-1],[1,1,1])

